url28f181064f_simonjupp_java_skos_api_tgz-pJ8-example_CreateSKOSExampleJ8/src/uk/ac/manchester/cs/skos/SKOSDatasetImpl.java

```diff
@@ -0,0 +1,533 @@
+import org.semanticweb.skosapibinding.SKOSManager;
+import org.semanticweb.skosapibinding.SKOStoOWLConverter;
+
+import java.net.URI;
+import java.util.*;
+/*
+ * Copyright (C) 2007, University of Manchester
+ *
+ * Modifications to the initial code base are copyright of their
+ * respective authors, or their employers as appropriate.  Authorship
+ * of the modifications may be determined from the ChangeLog placed at
+ * the end of this file.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+/**
+ * Author: Simon Jupp<br>
+ * Date: Mar 4, 2008<br>
+ * The University of Manchester<br>
+ * Bio-Health Informatics Group<br>
+ *
+ * A <code>SKOSDataset</code> is the central point of access to your SKOS vocabularies.
+ * The SKOS vocabulary is managed by a <code>SKOSManager</code>. A SKOSDataset can be thought
+ * of as an OWL Ontology which conatins a single of multiple SKOS concept schemes.<br>
+ * Because the SKOS API is built on top of the OWL API we use an OWLOntology to contain our SKOS Vocabularies,
+ * you always have access to the underlying OWLOntology, but in most cases you should be fine just using the SKOSDataset directly.
+ *
+ */
+public class SKOSDatasetImpl implements SKOSDataset {
+
+    private OWLOntologyManager man;
+    private SKOSManager skosManContent;
+    private OWLOntology owlOntology;
+    private Map<URI, SKOSEntity> SKOSEntity;
+
+    private SKOStoOWLConverter skos2owlConverter;
+
+    private BidirectionalShortFormProviderAdapter biAdapt;
+
+    /* Constructors */
+
+    public SKOSDatasetImpl(SKOSManager skosMan, OWLOntology onto) {
+        this.skosManContent = skosMan;
+        this.owlOntology = onto;
+        this.man = skosMan.getOWLManger();
+        this.SKOSEntity = new HashMap<URI, SKOSEntity>(20000);
+        skos2owlConverter = new SKOStoOWLConverter();
+//        this.propUtility = new SKOSObjectPropertyUtility(man);
+//        setUpShortFormProviders ();
+    }
+
+    public SKOSDatasetImpl(SKOSManager skosMan, URI uri) throws SKOSCreationException {
+        this.skosManContent = skosMan;
+        this.man = skosMan.getOWLManger();
+        try {
+            this.owlOntology = man.createOntology(IRI.create(uri));
+        } catch (OWLOntologyCreationException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)